### ISSUE TITLE
moon needed?

### ISSUE THREAD

**Issue by dneise** [fact-project/shiftcalendar#1]
Some say: 

We need "moon" or "planned" shutdown entries in the shiftcalendar so a consistency check can find out if all shifters are missing or if there is just no shift planned at all.

Others say:

If we have SSP shifts planned for a whole month, then it is no problem, that there is no observation planned during some of these days. The people just have nothing to do during these days, but they are technically on shift. So in case we will do something like scheduled technical measurements one day, the shifters are already there to perform the measurements.

**Comment by dneise** [comment]
@MaxNoe  What do you think? Is there a real reason for having moon shutdowns in the shiftcalendar, while they are already in the schedule in the sense, that there is no startup and no shutdown scheduled on a "moon"-day?

**Comment by maxnoe** [comment]
Yes, i think it's an important feature to see the difference between "Shifter Info missing" and "Moon break".

Also, the automatic schedule relies on this information:
https://trac.fact-project.org/browser/trunk/DataCheck/Monitoring/InsertSchedule.sh

```bash
   query="SELECT u FROM calendar.Data WHERE y="$year" AND m="$month"-1 AND d="$day" AND u NOT IN ('ETHZ','TUDO','ISDC','UNIWUE')"
   printprocesslog "DEBUG "$query
   #echo "DEBUG "$query
   shifters=`sendquery`
   if [ "$shifters" = "" ]
   then
       printprocesslog "INFO No shifter found in calendar -> Do not insert a schedule."
       echo "INFO No shifter found in calendar -> Do not insert a schedule."
       continue
   else
       checkshifter=`echo $shifters | grep moon`
       if [ "$checkshifter" != "" ]
       then
          printprocesslog "INFO Full moon night -> Do not insert a schedule."
          #echo "INFO Full moon night -> Do not insert a schedule."
          continue
       fi
   fi
```

**Comment by dneise** [comment]
Ah, thanks! it's that way around.

**Comment by dneise** [comment]
Ah, so the information when MAGIC has a moon shutdown is not freely available, so I (as a person) cannot get this information. I can read it from the old `factdb.calendar.Data` table (where it is until nov. 2017) but I can not further provide this data. 

For now I will read it from the old table.

**Comment by dneise** [comment]
Ah just another question. At the moment, the `calendarentry` table looks like this:

```
mysql> explain calendarentry;
+---------+----------+------+-----+---------+----------------+
| Field   | Type     | Null | Key | Default | Extra          |
+---------+----------+------+-----+---------+----------------+
| id      | int(11)  | NO   | PRI | NULL    | auto_increment |
| user_id | int(11)  | NO   |     | NULL    |                |
| role_id | int(11)  | NO   | MUL | NULL    |                |
| start   | datetime | NO   |     | NULL    |                |
| end     | datetime | NO   |     | NULL    |                |
+---------+----------+------+-----+---------+----------------+
```

For moon shut down nights, I will generate start and end time from sun set and dawn as usual. But what should I put for user_id and for role_id. Should I allow NULL as a value for `user_id`? since there is nobody on shift? But this opens up possibilities for errors. Same question for `role_id`. Should I generalte a pseudo role called "moon shutdown"? 

Alternatively I could just generate another table, which might look like this:
```
mysql> explain moonshutdown;
+---------+----------+------+-----+---------+----------------+
| Field   | Type     | Null | Key | Default | Extra          |
+---------+----------+------+-----+---------+----------------+
| id      | int(11)  | NO   | PRI | NULL    | auto_increment |
| date    | date     | NO   |     | NULL    |                |
+---------+----------+------+-----+---------+----------------+
```

Which contains all the moon shutdowns. They are not the same things as shift entries. So maybe pressing them into the table for the shiftentries (which I wrongly called "calendarentry", so that is sounds it can take in all kinds of calendarentries) is not a good idea. What do you think @MaxNoe ?

Still when presenting the stuff on the website, I have to combine both tables, but that is a question of presentation not of storing the data.... so ... well ... unsure ... first option seems to be less work, but wrong.

**Comment by maxnoe** [comment]
Why not a boolean `moon_break` field?

But an extra table for the moon breaks might give us more power.

**Comment by dneise** [comment]
| Why not a boolean moon_break field?

You mean an extra field in `calendarentry` that is called `moon_beak`? ... Hm let me think....
Wouldn't that still raise the same questions? What to put in `role_id` and `user_id` when `moon_break` is `True`? 
This feels to me like this mistake in OOP, where one combines two things in the same class and steers the behaviour of the class with an extra bool.... or even an integer called `type` or so. An example might be this:
https://trac.fact-project.org/browser/trunk/Mars/mhist/MHCamera.cc#L1273
